CesiumGeospatial/test/TestSimplePlanarEllipsoidCurve.cpp

```diff
@@ -0,0 +1,174 @@
+#include <CesiumGeospatial/SimplePlanarEllipsoidCurve.h>
+
+#include <catch2/catch.hpp>
+
+using namespace CesiumGeospatial;
+using namespace CesiumUtility;
+
+const glm::dvec3
+    philadelphiaEcef(1253264.69280105, -4732469.91065521, 4075112.40412297);
+const glm::dvec3
+    tokyoEcef(-3960158.65587452, 3352568.87555906, 3697235.23506459);
+
+// Values obtained from the Unreal implementation with all curves disabled.
```

Comment:
I'm generally not a fan of a unit test that checks itself based on an earlier implementation. What if we change the interpolation logic slightly? Would we update these values? How do we know the values are actually good?

From what I gather, we want to check "Is the flight path sensible in the middle part?". There are probably other ways to do this... ex. checking height never goes below 0 for a test with start / end with position heights. ex. check that the generated points actually lie on a plane (like the class suggest, etc...

What do you think?

Reply:
Removed the midpoint test, changed the "reverse path" test and "equivalent LLH coordinates" tests to check themselves against other curves instead of known values, and added tests for coplanarity and making sure the path doesn't cross through the earth.

Reply:
Looks great!